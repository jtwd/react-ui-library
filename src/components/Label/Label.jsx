import React from 'react'
import { string, bool } from 'prop-types'
import { Label as StyledLabel, RequiredSymbol } from './Label.styles'

function Label({ htmlFor, label, required }) {
  return (
    <StyledLabel htmlFor={htmlFor}>
      {label} {required && <RequiredSymbol title="Required field" />}
    </StyledLabel>
  )
}

Label.propTypes = {
  htmlFor: string.isRequired,
  label: string.isRequired,
  required: bool
}

Label.defaultProps = {
  required: false
}

export default Label
