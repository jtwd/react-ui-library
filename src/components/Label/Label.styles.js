import styled from 'styled-components'

import { xxs } from '../_theme/spacers'
import { red } from '../_theme/colors'
import { primaryBold } from '../_theme/units'
import pseudo from '../_theme/mixins/pseudo'

export const Label = styled.label`
  padding: ${xxs};
  display: block;
`

export const RequiredSymbol = styled.span`
  &::before {
    ${pseudo('inline-block', 'relative', '*')}
    color: ${red};
    font-weight: ${primaryBold};
  }
`
