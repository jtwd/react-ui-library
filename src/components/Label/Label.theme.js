import { getTheme } from '../_theme/utils'

const { spacing } = getTheme()

export default {
  padding: spacing.xxs
}
