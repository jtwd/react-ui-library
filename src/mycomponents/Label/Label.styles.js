import styled from 'styled-components'
import { xxs } from '../../theme/spacers'
import { red } from '../../theme/colors'
import { primaryBold } from '../../theme/units'

export const Label = styled.label`
  padding: ${xxs};
  display: block;
`

export const RequiredSymbol = styled.span`
  &:before {
    color: ${red};
    font-weight: ${primaryBold};
    display: inline-block;
    content: '*';
  }
`
