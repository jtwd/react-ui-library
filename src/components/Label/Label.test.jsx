import React from 'react'
import renderer from 'react-test-renderer'

import Label from './Label'

test('Label renders correctly', () => {
  const component = renderer.create(<Label htmlFor="test" label="test" />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Label renders required field when passed required prop', () => {
  const component = renderer.create(<Label htmlFor="text" label="test" required />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
