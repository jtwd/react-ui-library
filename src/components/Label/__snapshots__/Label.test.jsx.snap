// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<label
  className="sc-bdVaJa etThJd"
  htmlFor="test"
>
  test
   
</label>
`;

exports[`Label renders required field when passed required prop 1`] = `
<label
  className="sc-bdVaJa etThJd"
  htmlFor="text"
>
  test
   
  <span
    className="sc-bwzfXH hEsmVW"
    title="Required field"
  />
</label>
`;
